### PR TITLE
log errors in more detail when block production fails

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/spec/state_transition_block.nim [hunk 1]
```diff
@@ -342,33 +342,22 @@ proc process_operations(preset: RuntimePreset,
...
 
-  logScope:
-    blck = shortLog(blck)
   let res_block = process_block_header(state, blck, flags, stateCache)
   if res_block.isErr:
-    debug "Block header not valid",
-      block_header_error = $(res_block.error),
-      slot = state.slot
-    return false
+    return err("Block header not valid: " & $(res_block.error))
```
Comment (arnetheduck) [inline beacon_chain/spec/state_transition_block.nim:352]: why not just return `res_block` as-is?

File: beacon_chain/spec/state_transition.nim [hunk 2]
```diff
@@ -236,10 +244,15 @@ proc makeBeaconBlock*(
       voluntary_exits:
         List[SignedVoluntaryExit, Limit MAX_VOLUNTARY_EXITS](voluntaryExits)))
 
-  let ok = process_block(preset, state.data, blck, {skipBlsValidation}, cache)
+  let res = process_block(preset, state.data, blck, {skipBlsValidation}, cache)
```
Comment (arnetheduck) [inline beacon_chain/spec/state_transition.nim:247]:
```suggestion
  ? process_block(preset, state.data, blck, {skipBlsValidation}, cache)
```

File: beacon_chain/spec/state_transition_block.nim [hunk 3]
```diff
@@ -342,33 +342,22 @@ proc process_operations(preset: RuntimePreset,
...
     state: var BeaconState, blck: SomeBeaconBlock, flags: UpdateFlags,
-    stateCache: var StateCache): bool {.nbench.}=
+    stateCache: var StateCache): Result[void, string] {.nbench.}=
   ## When there's a new block, we need to verify that the block is sane and
   ## update the state accordingly - the state is left in an unknown state when
   ## block application fails (!)
 
-  logScope:
-    blck = shortLog(blck)
   let res_block = process_block_header(state, blck, flags, stateCache)
```
Comment (arnetheduck) [inline beacon_chain/spec/state_transition_block.nim:350]:
```suggestion
  ? process_block_header(state, blck, flags, stateCache)
```

Comment (tersec) [reply beacon_chain/spec/state_transition_block.nim:350]: `Result[void, cstring]` vs `Result[void, string]`

I can do the same `cstring` `const`-declaring dance as elsewhere, though.

File: beacon_chain/spec/state_transition_block.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/spec/state_transition_block.nim]: https://github.com/status-im/nimbus-eth2/pull/2135/commits/a37f650bb210f7836fbe4a8bb4ab74a0d1cef4f8

File: beacon_chain/spec/state_transition_block.nim (hunk 3 above)
Comment (arnetheduck) [reply beacon_chain/spec/state_transition_block.nim:350]: why not return `Result[void, cstring]`?

**Review by arnetheduck — APPROVED** [review]

File: beacon_chain/spec/state_transition_block.nim (hunk 3 above)
Comment (tersec) [reply beacon_chain/spec/state_transition_block.nim]: https://github.com/status-im/nimbus-eth2/pull/2135/commits/b36785192d3412e46fb19f6de68c588ac53257d7

**Pull request merged**